apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -707,6 +707,30 @@ where
 
         // Tx type check
         if let TxType::Wrapper(wrapper) = tx_type {
+            // Tx gas limit
+            let mut gas_meter = TxGasMeter::new(u64::from(&wrapper.gas_limit));
+            if gas_meter.add_tx_size_gas(tx_bytes.len()).is_err() {
+                response.code = ErrorCodes::TxGasLimit.into();
+                response.log =
+                    "Wrapper transactions exceeds its gas limit".to_string();
+                return response;
+            }
+
+            // Max block gas
```

Comment:
I think in the mempool we shouldn't check max block gas as that would prevent queuing up more txs than what fits into a single block

Reply:
In here I actually just check that the single tx in exam doesn't exceed the gas allowed for the entire block by itself. `prepare_proposal` will then construct the block computing the cumulated gas required by the txs that are being included

Reply:
I see, thx!